themes/gatsby-starter-theme/src/gatsby-theme-blog/components/header.js

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import { Link } from "gatsby"
+import { css, Styled } from "theme-ui"
+import Header from "gatsby-theme-blog/src/components/header"
+
+export default props => (
+  <Header {...props} title="Custom Title">
```

Comment:
The `title` prop here changes what the starter shows – should I change it back to say `Shadowed Site Title`, @amberleyromo?

Reply:
Okay, dug into this again, and I have a question: the notes layout does not render the `title` from `siteMetadata` because that query is currently in the blog theme's template. How do we want to handle component-level queries like this?

Reply:
can the notes theme make its own query? it's querying the same config, so it would still have `title` in `siteMetadata` on the site level.

Reply:
Why not put the `StaticQuery` in the header itself?

Reply:
If the query is in the header component, then it'd be on the user's end to also include that in the shadowed component – that seems a little less-than-ideal. If we make all official themes follow the same siteMetadata setup, that seems like it'd be a case of it works here, but not with other themes

Reply:
I think that makes sense though. These are built to work together -- if you're composing another theme in that wasn't built for that, it makes sense that you might need to do more manual work to tune it. But these were designed to work together.

Reply:
> If the query is in the header component, then it'd be on the user's end to also include that in the shadowed component – that seems a little less-than-ideal.

That seems pretty ideal to me. If I don't want the data in my header then the entire query goes away, and this also means it's not passed through other arbitrary components that don't care about the data like `Layout`.

If you're shadowing a component you can copy the whole file over, which gives you the data and the component in a single location rather than having to learn how page context works.